schema_editor/app/scripts/userdata/user-service.js

```diff
@@ -42,27 +44,52 @@
 
         function getUser(userId) {
             var dfd = $q.defer();
-            var result = module.User.get({id: userId}, function () {
-                dfd.resolve(result);
+            module.User.get({id: userId}, function (user) {
+                if (user && user.groups) {
+                    // append attribute to response to indicate if user is an admin or not
+                    user.isAdmin = (user.groups.indexOf(ASEConfig.api.groups.admin) > -1) ? true : false;
+                }
+                dfd.resolve(user);
             });
             return dfd.promise;
         }
 
+        // Check whether user has write access
+        function canWriteRecords(userId, token) {
+            tmpToken = token;
+            var dfd = $q.defer();
+            module.User.query({id: userId}, function (user) {
+                if (user && user.groups) {
+                    // admin or analyst can write records
+                    if (user.groups.indexOf(ASEConfig.api.groups.admin) > -1 ||
+                        user.groups.indexOf(ASEConfig.api.groups.readWrite) > -1) {
+
+                        dfd.resolve(true);
+                    } else {
+                        dfd.resolve(false);
+                    }
+                } else {
+                    dfd.resolve(false);
+                }
+                tmpToken = '';
+            });
+
+            return dfd.promise;
+        }
+
+        // Check whether user is an admin or not before logging them in (for the editor)
         function isAdmin(userId, token) {
             tmpToken = token;
             var dfd = $q.defer();
             module.User.query({id: userId}, function (user) {
-                /* jshint camelcase: false */
-                if (user && user.is_staff) {
+                if (user && user.groups && user.groups.indexOf(ASEConfig.api.groups.admin) > -1) {
```

Comment:
This check shows up in a few different places, it might shorten things up a bit to pull it out into a function.